### ISSUE TITLE
value return by getViewportWidthInCssPixels() is wrong on retina display

### ISSUE THREAD

**Issue by thomaspaillot** [tubalmartin/riloadr#20]
I was trying to make a special breakpoint for big retina displays (iPad 3, macbook pro retina…) but riloadr doesn't seem to take that breakpoint and instead use my iphone4 breakpoint.

After some debug I found out that the variable screenWidthFallback on the iPad 3 return 384 (768 return by window.screen.width divided by pixel ratio).

When using the meta 'viewport' with width=device-width window.screen.width returns the ideal viewport size already divided by pixel ratio, so on iPad 3 and macbook retina getViewportWidthInCssPixels() returns a wrong width value.

Maybe I miss something.

By the way great library !

**Comment by mdoyle13** [comment]
I also came across this issue. I'm not a super skilled javascript developer but I fixed my issue (at least on a macbook pro retina). I just made sure that screenWidthFallback doesn't divide by the pixel ratio. I should also mention I have this in my document head. 

```
<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
```

Also please let me know what issues you guys see with this. I'm guessing it's not a perfect solution!

New function looks like

``` javascript
function getViewportWidthInCssPixels() {
        var math = Math
          , widths = [docElm.clientWidth, docElm.offsetWidth, body.clientWidth]
          , screenWidthFallback = math.ceil(screenWidth) //doesn't divide by devicePixelRatio
          , l = widths[LENGTH]
          , i = 0
          , width;

        for (; i < l; i++) {
            // If not a number remove it
            if (isNaN(widths[i])) {
                widths.splice(i, 1);
                i--;
            }
        }

        if (widths[LENGTH]) {
            width = math.max[APPLY](math, widths);

            // Catch cases where the viewport is wider than the screen
            if (!isNaN(screenWidthFallback)) {
                width = math.min(screenWidthFallback, width);
            }
        }

        return width || screenWidthFallback || 0;
    }

```

**Comment by thomaspaillot** [comment]
Thanks for the update, I'm going to check on an iPad retina and on a macbook retina. I'll let you know what I find.

**Comment by tubalmartin** [comment]
Thanks for the report! 

Could you try loading [this page](http://tubalmartin.me/riloadr/test.html) in every device you own to check if the output of getViewportWidthInCssPixels() equals the expected value?

**Comment by mdoyle13** [comment]
Hey Guys,

I have a regular monitor, a macbook with retina display and an iPhone with
retina display and they all reported the correct data.

**Comment by tubalmartin** [comment]
Fixed in 1.5.1! Thank you for reporting!
